src/bosh_azure_cpi/lib/cloud/azure/restapi/azure_client.rb

```diff
@@ -123,11 +123,47 @@ def rest_api_url(resource_provider, resource_type, resource_group_name: nil, nam
       url
     end
 
+    # get single resource
+    # example: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/networkInterfaces/{networkInterfaceName}
     def get_resource_by_id(url, params = {})
       result = nil
       begin
-        result = http_get(url, params)
+        uri = http_url(url, params)
+        response = http_get(uri)
+        result = JSON.parse(response.body) unless response.body.nil?
       rescue AzureNotFoundError => e
+        @logger.debug("Resource not found for url #{url} with parms #{params}")
+        result = nil
+      end
+      result
+    end
+
+    # get list of resources
+    # example: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/networkInterfaces
+    def get_resources_by_url(url, params = {})
```

Comment:
do we need to add unit test in this PR?

Reply:
yes, we should. but busy on other things. however, this is already tested manually.